client/failing_funding_test.go

```diff
@@ -0,0 +1,108 @@
+// Copyright 2022 - See NOTICE file for copyright holders.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package client_test
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"math/big"
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+
+	"perun.network/go-perun/channel"
+	chtest "perun.network/go-perun/channel/test"
+	"perun.network/go-perun/client"
+	ctest "perun.network/go-perun/client/test"
+	"perun.network/go-perun/wire"
+	"polycry.pt/poly-go/test"
+)
+
+func TestFailingFunding(t *testing.T) {
+	const (
+		timeout      = 1 * time.Second
+		fridaIdx     = 0
+		fredIdx      = 1
+		fridaInitBal = 100
+		fredInitBal  = 50
+	)
+
+	ctx, cancel := context.WithTimeout(context.Background(), timeout)
+	defer cancel()
+	rng := test.Prng(t)
+
+	setups := NewSetups(rng, []string{"Frida", "Fred"})
+	// Inject the failing funder for Frida.
+	setups[fridaIdx].Funder = FailingFunder{}
```

Comment:
It might be nice to test whether it also works if Fred has a failing funder.

Reply:
Done.